v2/kotlin-sdk/app/src/main/java/com/mongodb/app/ItemActivity.kt

```diff
@@ -0,0 +1,482 @@
+                // Disallow adding tasks when only filtering by completed tasks
+//                fab.isEnabled = !isChecked
+
+                CoroutineScope(Dispatchers.IO).launch {
+                    realm.subscriptions.update {
+                        val queryMine = realm.query<Item>("owner_id == $0", realmApp.currentUser!!.id)
+                        val queryMineCompleted = realm.query<Item>("owner_id == $0 && isComplete == $1", realmApp.currentUser!!.id, true)
+
+                        if (isChecked) {
+                            removeAll()
+                            queryMineCompleted.subscribe(SUBSCRIPTION_MINE_COMPLETED)
+                        } else {
+                            removeAll()
+                            queryMine.subscribe(SUBSCRIPTION_MINE)
+                        }
+                    }
+                    realm.subscriptions.waitForSynchronization(10.seconds)
+                }
+            }
+        }
+    }
+
+    /**
+     * Add buttons to the item menu.
+     */
+    override fun onCreateOptionsMenu(menu: Menu): Boolean {
+        menuInflater.inflate(R.menu.item_menu, menu)
+        airplaneModeItem = menu.findItem(R.id.action_airplane_mode)
+
+        if (this::realm.isInitialized) {
+            if (connectivityEnabled) {
+                airplaneModeItem.icon = syncEnabledIcon
+                realm.syncSession.resume()
+            } else {
+                airplaneModeItem.icon = syncDisabledIcon
+                realm.syncSession.pause()
+            }
+        }
+
+        return true
+    }
+
+    /**
+     *  Decides actions for all buttons on the item menu.
+     *  When "log out" is tapped, logs out the user and brings them back to the login screen.
+     */
+    override fun onOptionsItemSelected(menuItem: MenuItem): Boolean {
+        return when (menuItem.itemId) {
+            R.id.action_logout -> {
+                CoroutineScope(Dispatchers.IO).launch {
+                    runCatching {
+                        realmApp.currentUser?.logOut()
+                    }.onSuccess {
+                        Log.v(TAG(), "user logged out")
+                        withContext(Dispatchers.Main) {
+                            startActivity(Intent(application, LoginActivity::class.java))
+                        }
+                    }.onFailure { ex: Throwable ->
+                        Log.e(TAG(), "log out failed! Error: ${ex.message}")
+                    }
+                }
+                true
+            }
+            R.id.action_airplane_mode -> {
+                if (connectivityEnabled) {
+                    connectivityEnabled = false
+                    airplaneModeItem.icon = syncDisabledIcon
+                    realm.syncSession.pause()
+                } else {
+                    connectivityEnabled = true
+                    airplaneModeItem.icon = syncEnabledIcon
+                    realm.syncSession.resume()
```

Comment:
[minor] Is there an easy way to tie the state of the button (offline/online) with the actual state of the realm syncSession (paused/resumed)? It may be less prone to state mismatch from input spam that way.

Reply:
Legacy code, it has been removed.